### PR TITLE
Fix URL redirect for go back link

### LINKED ISSUE #9688: go back where you came from link doesn't do anything
### Problem

I'm on mobile so can't investigate more but the link doesn't do anything and I can't even copy it. Not sure what's up with that.

#### Relevant URL(s)
https://openlibrary.org/works/OL264247W

### Reproducing the bug

1. Go to ...
2. Do ...

* Expected behavior:
* Actual behavior:

### Context

- Browser (Chrome, Safari, Firefox, etc):
- OS (Windows, Mac, etc):
- Logged in (Y/N): Y
- Environment (prod, dev, local): prod

#### Stakeholders

<hr>

#### Instructions for Contributors

- Please [run these commands](https://github.com/internetarchive/openlibrary/wiki/Git-Cheat-Sheet#working-on-your-branch) to ensure your repository is up to date **before** [creating a new branch](https://github.com/internetarchive/openlibrary/wiki/Git-Cheat-Sheet#making-changes-and-creating-a-pull-request) to work on this issue and **each time after** pushing code to Github, because the pre-commit bot may add commits to your PRs upstream.

### PR DESCRIPTION
**Pull request by schu96**
Closes #9688

Fix

### Testing
I deleted an entry until the "This page has been deleted" page was shown. I checked that the previous expected behavior worked, then copied and pasted the entry's URL into a new browser window to check if the 'go back' link would redirect to the OpenLibrary homepage.

### Stakeholders
@jimchamp

### CODE REVIEW

**Comment by mekarpeles** [comment]
We may have (a) a few style tweaks and (b) numerical formatting to make to our search results + star ratings PR, though overall looking good. And the half star and whole stars should be the same gold color

image

perhaps we're missing a style rule for reader-stats__star. --half currently has the style font-size: 1.5em; which should probably be  removed (the size should be on whatever the base class is that --half overwrites, and this base class also needs the right @gold ffd400 color

**Comment by schu96** [comment]
@mekarpeles Is your comment for intended for this [PR](https://github.com/internetarchive/openlibrary/pull/9790)?

File: openlibrary/templates/type/delete/view.html [hunk 1]
```diff
@@ -12,7 +12,7 @@ <h1>$_("This page has been deleted")</h1>
     <p><strong>$_("What would you like to do?")</strong></p>
 
     <ul>
-      <li><a href="javascript:;" onclick="history.go(-1);">$_("Go back where I came from")</a></li>
+      <li><a href="javascript:;" id="go-back">$_("Go back where I came from")</a></li>
```
Comment (jimchamp) [inline openlibrary/templates/type/delete/view.html:15]:
```suggestion
      <li><a href="javascript:;" class="go-back-link">$_("Go back where I came from")</a></li>
```
There are a number of these links throughout the codebase.  I think that it's best to identify them with a common class, then add the click listener to all elements having this class.

Doing a quick search for `history.go` with my IDE yielded 14 matches.  I'd expect all of these to be updated to use your click listener in this branch.

File: openlibrary/plugins/openlibrary/js/index.js [hunk 2]
```diff
@@ -562,5 +562,12 @@ jQuery(function () {
         import(/* webpackChunkName: "fulltext-search-suggestion" */ './fulltext-search-suggestion')
             .then(module => module.initFulltextSearchSuggestion(fulltextSearchSuggestion))
     }
+
+    // Go back redirect:
+    const goback = document.querySelector('#go-back')
+    if (goback) {
+        import (/* webpackChunkName: "delete-redirect" */ './delete-redirect')
+            .then(module => module.initGoBackRedirect(goback))
```
Comment (jimchamp) [inline openlibrary/plugins/openlibrary/js/index.js:570]:
```suggestion
    const backLinks = document.querySelectorAll('.go-back-link')
    if (backLinks.length) {
        import (/* webpackChunkName: "go-back-links" */ './go-back-links')
            .then(module => module.initGoBackLinks(backLinks))
```
A couple of changes here:
1. Select all elements on the page that have the `go-back-link` class, then pass them to the `init` method (which should be updated to handle a [`NodeList`](https://developer.mozilla.org/en-US/docs/Web/API/NodeList) of elements).
2. Update the name of the `init` function, and the `delete-redirect.js` file name.  When I see the name `delete-redirect.js`, it's not clear that the code is related to the "Go Back" links.  I expect that naming this something more obvious will make the code easier to maintain (especially as my memory of this PR fades over time).

**Review by jimchamp — CHANGES_REQUESTED** [review]
Thanks, @schu96!
This is a great start.  Please check out my review notes, and apply your new click handler to all "Go Back" links.

**Review by jimchamp — APPROVED** [review]
Seems to be working as expected.  Thanks, @schu96!

**Pull request merged**